src/main/kotlin/org/neo4j/graphql/Translator.kt

```diff
@@ -86,12 +86,74 @@ class Translator(val schema: GraphQLSchema) {
                            "WITH $variable as toDelete, ${mapProjection.query} AS $variable $ordering$skipLimit DETACH DELETE toDelete RETURN $variable",
                                 (mapProjection.params + mapOf(paramName to properties.params[paramName])))
                     }
-                    else -> throw IllegalArgumentException("Unknown Mutation "+name)
+                    else -> checkRelationships(fieldDefinition, field, ordering, skipLimit, ctx)
                 }
             }
         }
     }
 
+    private fun checkRelationships(graphQLFieldDefinition: GraphQLFieldDefinition, field: Field, ordering: String, skipLimit: String, ctx: Context): Cypher {
+        val source = graphQLFieldDefinition.type as GraphQLObjectType
+        val targetPair = source.fieldDefinitions
+                .filter {
+                    field.name == "add${source.name}${it.name.orEmpty().capitalize()}"
+                            || field.name == "delete${source.name}${it.name.orEmpty().capitalize()}"
+                }
+                .map { it.name to it.type }
+                .firstOrNull() ?: throw IllegalArgumentException("Unknown Mutation ${graphQLFieldDefinition.name}")
+
+        val isList = targetPair.second.isList()
+        val target = if (isList) {
+            (targetPair.second as GraphQLList).wrappedType as GraphQLObjectType
+        } else {
+            targetPair.second as GraphQLObjectType
+        }
+        val targetFieldName = targetPair.first
+
+        val sourceId = source.fieldDefinitions.find { it.type.inner() == Scalars.GraphQLID }
+                ?: throw IllegalArgumentException("Unknown Mutation ${graphQLFieldDefinition.name}")
```

Comment:
is this still an unknown mutation?

Reply:
no is managed by the `augmentedSchema` method.
Removed